Sources/AsyncConnection.swift

```diff
@@ -3,7 +3,7 @@ public protocol AsyncConnection: AsyncStream {
 }
 
 extension AsyncConnection {
-    public func open(result: (Void throws -> AsyncConnection) -> Void) throws {
+    public func open(_ result: (Void throws -> AsyncConnection) -> Void) throws {
```

Comment:
I'm assuming once we have generic type aliases we'll just have something like
`typealias Completion<T> = (Void throws -> T) -> Void`

Also, I'm not a huge fan of that closure anyway - it's really hard to parse without seeing a good example or thinking about it for a while first. Perhaps we should make a wrapper around it (and then we can take advantage of generics)? LMK if this is something that I should raise discussion about separately.